InscryptionAPI/Triggers/Interfaces.cs

```diff
@@ -1,240 +1,414 @@
+        /// <returns>The priority for this trigger.</returns>
+        public int GetTriggerPriority(PlayableCard card, OpposingSlotTriggerPriority modType, List<CardSlot> originalSlots, List<CardSlot> currentSlots, int attackCount, bool didRemoveDefaultSlot);
+    }
+
+    /// <summary>
+    /// Flag that tells IGetOpposingSlots when to trigger.
+    /// </summary>
+    public enum OpposingSlotTriggerPriority
+    {
+        /// <summary>
+        /// Normal mode - use this for opposing slot modifications that either don't do anything to the original opposing slots or remove it.
+        /// </summary>
+        Normal,
+        /// <summary>
+        /// Bring back opposing slot - use this for opposing slot modifications that bring back the original opposing slot if it was removed like Trifurcated Strike.
+        /// </summary>
+        BringsBackOpposingSlot,
+        /// <summary>
+        /// Post slot addition modification - use this for opposing slot modifications that don't interact with any other modification and instead just statically modify the finished result like Double Strike.
+        /// </summary>
+        PostAdditionModification,
+        /// <summary>
+        /// Replaces default opposing slot - use this for opposing slot modifications that replace the original opposing slot like Omni Strike.
+        /// </summary>
+        ReplacesDefaultOpposingSlot
+    }
+
+    /// <summary>
+    /// Data collection trigger that collects data about the usability of an item.
+    /// </summary>
+    public interface IItemCanBeUsed
+    {
+        /// <summary>
+        /// Returns true if this should modify if an item can be used.
+        /// </summary>
+        /// <param name="itemname">The name of the item that is about to get used.</param>
+        /// <param name="currentValue">True if it can be used currently.</param>
+        /// <returns>True if this should modify if an item can be used.</returns>
+        public bool RespondsToItemCanBeUsed(string itemname, bool currentValue);
+        /// <summary>
+        /// Returns true if an item can be used.
+        /// </summary>
+        /// <param name="itemname">The name of the item that is about to get used.</param>
+        /// <param name="currentValue">True if it can be used currently.</param>
+        /// <returns>True if the item can be used, false otherwise.</returns>
+        public bool CollectItemCanBeUsed(string itemname, bool currentValue);
+    }
+
+    /// <summary>
+    /// Trigger that is triggered when an item is prevented from use using IItemCanBeUsed.
+    /// </summary>
+    public interface IOnItemPreventedFromUse
+    {
+        public bool RespondsToItemPreventedFromUse(string itemname);
+        public IEnumerator OnItemPreventedFromUse(string itemname);
```

Comment:
Please camelCase and add some documenation. I know the name itself is straightforward, but will be nice to have consistency.